### ISSUE TITLE
Naming clash with Joda convert classes

### ISSUE THREAD

**Issue by marko-asplund** [elastic/elasticsearch#10290]
elasticsearch.jar (v1.5.0) re-packages Joda convert classes:

* org.joda.convert.FromString
* org.joda.convert.ToString

this will create a naming clash in projects that use Joda convert.

**Comment by Pyppe** [comment]
Same issue here. Cannot create our own packaged application, because elasticsearch.jar contains the above-mentioned joda-classes.

**Comment by marko-asplund** [comment]
here's how to reproduce this issue:

```
wget https://dl.bintray.com/sbt/native-packages/sbt/0.13.8/sbt-0.13.8.zip
unzip sbt-0.13.8.zip
git clone https://github.com/marko-asplund/es-repro
cd es-repro
../sbt/bin/sbt assembly
```

Assembly fails with ES 1.5.0. However, if you edit `build.sbt` in the repository to replace the ES 1.5.0 dependency with ES 1.4.4 assembly is performed successfully.

**Comment by fuwaneko** [comment]
Very annoying bug effectively preventing us from creating fat jar in our project. Looking back to original issue #4660 that introduced monkey-patched files from joda-convert into ES I can only sigh at how developers sometimes do things purely wrong and forget about it until it blows up someday.

I can't even use usual sbt/sbt-assembly stuff to exclude these files during build. That's unacceptable.

**Comment by pheaver** [comment]
Yup, I'm also unable to build a package because of this.

**Comment by s1monw** [comment]
can you guys use the shaded jar? 

```xml
 <groupId>org.elasticsearch.distribution.shaded</groupId>
 <artifactId>elasticsearch</artifactId>
```

**Comment by pheaver** [comment]
Oh, I'm not familiar with that. Do you know how I would use that in sbt?

**Comment by s1monw** [comment]
I don't know sbt but how do you use elasticsearch?

**Comment by s1monw** [comment]
`libraryDependencies += "org.elasticsearch.distribution.shaded" % "elasticsearch" % "2.0.0-beta1"` according to http://search.maven.org/#artifactdetails%7Corg.elasticsearch.distribution.shaded%7Celasticsearch%7C2.0.0-beta1%7Cjar

**Comment by pheaver** [comment]
So the shaded package is only available as 2.x? I'm not sure if that will work.

Also, I'm using https://github.com/sksamuel/elastic4s, which maybe has a dependency on the non-shaded elasticsearch, so even when I try the shaded package I still get errors about conflicting versions of elasticsearch.

**Comment by s1monw** [comment]
> So the shaded package is only available as 2.x? I'm not sure if that will work.

yeah the 1.x dev branch is closed we are mainly focusing on 2.x now

**Comment by m4dc4p** [comment]
2.0.0 isn't even out of beta. Please fix this issue. It's super annoying. Just do a point release! (1.7.2)

**Comment by clintongormley** [comment]
Closed by #13358
